modin/core/execution/ray/implementations/pandas_on_ray/partitioning/virtual_partition.py

```diff
@@ -0,0 +1,487 @@
+# Licensed to Modin Development Team under one or more contributor license agreements.
+# See the NOTICE file distributed with this work for additional information regarding
+# copyright ownership.  The Modin Development Team licenses this file to you under the
+# Apache License, Version 2.0 (the "License"); you may not use this file except in
+# compliance with the License.  You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software distributed under
+# the License is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF
+# ANY KIND, either express or implied. See the License for the specific language
+# governing permissions and limitations under the License.
+
+"""Module houses classes responsible for storing an axis partition and applying a function to it."""
+
+import pandas
+
+from modin.core.dataframe.pandas.partitioning.axis_partition import (
+    PandasDataframeAxisPartition,
+)
+from .partition import PandasOnRayDataframePartition
+
+import ray
+from ray.util import get_node_ip_address
+
+
+class PandasOnRayDataframeVirtualPartition(PandasDataframeAxisPartition):
```

Comment:
Since this assumes its axis-partitioned, should axis be in its name?

Reply:
I think we will eventually change all `axis` to `virtual`, but this is trying to be a minimal PR.